pom.xml

```
@@ -4,32 +4,206 @@
+            <version>${commons-lang3.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-api</artifactId>
+            <version>${log4j.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-core</artifactId>
+            <version>${log4j.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-catalog-api</artifactId>
+            <version>${log4j-audit.version}</version>
+        </dependency>
+    </dependencies>
+
+    <!-- handle the maven build lifecycle phases -->
+    <build>
+        <plugins>
+            <!-- handles creation of jenkins plugin packages -->
+            <plugin>
+                <groupId>org.jenkins-ci.tools</groupId>
+                <artifactId>maven-hpi-plugin</artifactId>
+                <version>${maven-hpi-plugin.version}</version>
+            </plugin>
+            <!-- handle java class compilations -->
+            <!-- <plugin> <groupId>org.apache.maven.plugins</groupId> <artifactId>maven-compiler-plugin</artifactId> <version>3.6.1</version> <configuration> <source>1.8</source> <target>1.8</target> <showDeprecation>true</showDeprecation>
+            <showWarnings>true</showWarnings> <compilerArguments> <Xmaxwarns>10000</Xmaxwarns> <Xlint/> </compilerArguments> <encoding>UTF-8</encoding> </configuration> </plugin> -->
+
+            <!-- handle unit testing of java classes -->
+            <!-- <plugin> <groupId>org.apache.maven.plugins</groupId> <artifactId>maven-surefire-plugin</artifactId> <version>2.18.1</version> <executions> <execution> <id>default-test</id> <goals> <goal>test</goal> </goals> </execution> </executions>
+            <configuration> <includes> <include>**/Test*.java</include> <include>**/*Test.java</include> </includes> <excludes> <exclude>**/*FuncTest.java</exclude> </excludes> <forkCount>1</forkCount> <reuseForks>false</reuseForks> <systemPropertyVariables>
+            <environment>${environment}</environment> <site>${site}</site> </systemPropertyVariables> </configuration> </plugin> -->
+
+            <!-- log4j audit maven plugin -->
+            <plugin>
+                <groupId>org.apache.logging.log4j</groupId>
+                <artifactId>log4j-audit-maven-plugin</artifactId>
+                <version>${log4j-audit.version}</version>
+                <executions>
+                    <execution>
+                        <id>generate</id>
+                        <phase>generate-sources</phase>
+                        <goals>
+                            <goal>generate</goal>
+                        </goals>
+                        <configuration>
+                            <packageName>org.apache.logging.log4j.audit.event</packageName>
```

```suggestion
                            <packageName>io.jenkins.plugins.audit</packageName>
```

This needs to match the package you put the `RequestContext` class. Or you need to change this to something like `io.jenkins.plugins.audit.event` and add an `import io.jenkins.plugins.audit.event.*;` to `RequestContext.java`.

---

alright, I went with the first option.